core/src/main/kotlin/com/rudderstack/sdk/kotlin/core/internals/policies/backoff/ExponentialBackOffPolicy.kt

```diff
@@ -0,0 +1,55 @@
+package com.rudderstack.sdk.kotlin.core.internals.policies.backoff
+
+import com.rudderstack.sdk.kotlin.core.internals.utils.InternalRudderApi
+import java.security.SecureRandom
+import kotlin.math.pow
+
+internal const val MIN_INTERVAL = 10L
+internal const val MAX_INTERVAL = 60_000L
+internal const val DEFAULT_INTERVAL = 3000L
+
+internal const val MIN_BASE = 1.1
+internal const val MAX_BASE = 5.0
+internal const val DEFAULT_BASE = 2.0
+
+/**
+ * This class implements an exponential backoff strategy with jitter for handling retries.
+ * It allows for configurable interval and base for the exponential calculation.
+ * The next delay is calculated using the formula: delay = interval * base^attempt.
+ * The delay is then adjusted with a random jitter to avoid synchronized retries.
+ */
+@InternalRudderApi
+class ExponentialBackOffPolicy(
+    private var intervalInMillis: Long = DEFAULT_INTERVAL,
+    private var base: Double = DEFAULT_BASE,
+) : BackOffPolicy {
+    private var attempt = 0
+    private val random = SecureRandom()
+
+    init {
+        intervalInMillis = when {
+            intervalInMillis in MIN_INTERVAL..MAX_INTERVAL -> intervalInMillis
+            else -> DEFAULT_INTERVAL
+        }
+        base = when {
+            base in MIN_BASE..MAX_BASE -> base
+            else -> DEFAULT_BASE
+        }
+    }
+
+    override fun nextDelayInMillis(): Long {
+        val delayInMillis = (intervalInMillis * base.pow(attempt++)).toLong()
+        val delayWithJitterInMillis = withJitter(delayInMillis)
+
+        return delayWithJitterInMillis
```

Comment:
First add this:
```
// Define a top level constant:
private const val MILLISECONDS_IN_SECOND = 1000

// Define a top level function
private fun Int.toMilliSeconds(): Long = this * MILLISECONDS_PER_SECOND
```

```suggestion
        val delayInSec = minDelayInSec * base.pow(attempt++)
        val delayWithJitterInSec: Int = withJitter(delayInSec) // This can be made more readable with the use of infix function. But I'll skip it for now.

        return delayWithJitterInSec.toMilliseconds()
```

Reply:
I've not tested the proposed solution. Do the needful to make it work.